### PR TITLE
[UNR-795] Engine Versioning

### PR DESCRIPTION
**Pull request by jnicholas-io**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added a compile time check to ensure the engine version is compatible with the GDK. When breaking changes are made to the engine that make the GDK incompatible, the `SPATIAL_ENGINE_VERSION` in the engine is incremented. The corresponding `SPATIAL_GDK_VERSION` in the GDK is incremented alongside the engine version when changes have been made to the GDK to restore compatibility.

Corresponding Engine PR: https://github.com/improbableio/UnrealEngine/pull/123

#### Release note
Feature: Added a compile time check to ensure the GDK is compatible with the version of Unreal Engine being used

#### Primary reviewers
@m-samiec @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h]: Missing copyright header.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -31,6 +31,7 @@
 #include "SpatialConstants.h"
 #include "SpatialGDKSettings.h"
 #include "Utils/EntityPool.h"
+#include "Utils/EngineVersionCheck.h"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: nit: alphabetize includes

File: SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Launch/Resources/SpatialVersion.h"
+
+// GDK Version to be updated with SPATIAL_ENGINE_VERSION
+// when breaking changes are made to the engine that requires
+// changes to the GDK to remain compatible
+#define SPATIAL_GDK_VERSION 0
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h]: I know we're all programmers, but I feel version should start with 1 when there's no minor version.

Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Public/Utils/EngineVersionCheck.h]: please start at -1

**Pull request merged**